packages/gatsby/src/schema/graphql-engine/print-plugins.ts

```diff
@@ -199,16 +192,16 @@ function findSubPlugins(
 
         return []
       })
-      .map(plugin => plugin[`name`])
+      .map(plugin => plugin[`resolve`])
       .filter((p: unknown): p is string => typeof p === `string`)
   )
   return allFlattenedPlugins.filter(
-    p => usedSubPluginNames.has(p.name) && !!p.modulePath
+    p => usedSubPluginResolves.has(p.resolve) && !!p.modulePath
```

Comment:
Apparently `name` is not reliable as some sub-plugins may have no `name` (despite [our types](https://github.com/gatsbyjs/gatsby/blob/7461c7abead2a1cbd3fa8dff1fdb2e18ca0b9a33/packages/gatsby/src/redux/types.ts#L233))